pandora-server-directory/src/database/mongoDb.ts

```diff
@@ -255,15 +255,14 @@ export default class MongoDatabase implements PandoraDatabase {
 	}
 
 	public async queryAccountDisplayNames(query: AccountId[]): Promise<Record<AccountId, string>> {
-		// TODO get the actual display name when it's implemented
 		const result: Record<AccountId, string> = {};
 		const accounts = await this._accounts
 			.find({ id: { $in: query } })
-			.project<Pick<DatabaseAccount, 'id' | 'username'>>({ id: 1, username: 1 })
+			.project<Pick<DatabaseAccount, 'id' | 'settingsLimited'>>({ id: 1, settingsLimited: 1 })
 			.toArray();
 
 		for (const acc of accounts) {
-			result[acc.id] = acc.username;
+			result[acc.id] = acc.settingsLimited.displayName.value;
 		}
 		return result;
 	}
```

Comment:
This needs to default to username if the display name is empty, as we shouldn't allow empty names anywhere (because if the `settingsLimited` gets invalid somehow, they get reset into default value which _is_ empty in the current implementation (which is a bug as after reset it will immediately fail the validation, because that is not a valid value)).